src/test/java/org/folio/rest/support/WebClient.java

```diff
@@ -53,34 +55,34 @@ public void post(URL url,
     Object body,
     String tenantId,
     String userId,
-    Handler<HttpClientResponse> responseHandler) {
+    Handler<AsyncResult<HttpResponse<Buffer>>> responseHandler) {
 
-    HttpClientRequest request = client.postAbs(url.toString(),
-      responseHandler);
+    HttpRequest<Buffer> request = client.postAbs(url.toString());
 
     request.headers().add("Accept", "application/json, text/plain");
     request.headers().add("Content-type", "application/json");
 
     stdHeaders(request, url, tenantId, userId);
-    if (body != null) {
-      String encodedBody = Json.encodePrettily(body);
-      log.info(String.format("POST %s, Request: %s",
-        url.toString(), encodedBody));
-      request.end(encodedBody);
+
+    if (body == null) {
+      request.send(responseHandler);
     } else {
-      request.end();
+      Buffer encodedBody = Buffer.buffer(Json.encodePrettily(body));
```

Comment:
Using the guard clause pattern you can avoid the else clause:

    if (body == null) {
      request.send(responseHandler);
      return;
    }
    Buffer encodedBody = Buffer.buffer(Json.encodePrettily(body));
    ...